modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobWorker.java

```diff
@@ -993,6 +994,23 @@ else if (!internal && ctx.event().isRecordable(EVT_JOB_FAILED))
         }
     }
 
+    /**
+     * This method wraps U.error invocations to check node stopping.
+     * Log message will be skipped if node is stopping and debug is disabled.
+     *
+     * @param msg Message to log using quiet logger.
+     * @param e Optional exception.
+     */
+    private void logError(String msg, @Nullable Throwable e) {
+        if (e != null) {
```

Comment:
Can be rephrased as if (e != null && (log.isDebugEnabled() || !X.hasCause(e, NodeStoppingException.class)))